### ISSUE TITLE
Craft 3.1.19 breaks entriesConnection, et al.

### ISSUE THREAD

**Issue by timkelty** [markhuot/craftql#248]
Seems odd for a patch release, but queries with `entriesConnection` / `tagsConnection` / `categoriesConnection` started breaking upon upgrading `craftcms/cms` from `3.1.18 => 3.1.19`.

I locked the version down and confirmed it works if I go back to `3.1.18`.

Given query:

```
query {
  categoriesConnection(limit:1) {
    edges {
      node {
        id
      }
    }
  }
}
```

on `craftcms/cms@3.1.9`:
```
{
  "errors": [
    {
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "entriesConnection"
      ]
    }
  ],
  "data": {
    "entiresConnection": null
  }
}
```

on `craftcms/cms@3.1.8`:
```
{
  "data": {
    "entriesConnection": {
      "edges": [
        {
          "node": {
            "id": 18881
          }
        }
      ]
    }
  }
}
```

I haven't peeked at the source of the `3.1.19` release yet, but maybe @brandonkelly might be able to shine some light.

**Comment by timkelty** [comment]
I've tried this with `markhuot/craftql@1.3.1` and `dev-master` with the same result.

**Comment by brandonkelly** [comment]
3.1.19 was a pretty tame release. Just looked through it and not really sure what could be messing with CraftQL. Happy to look into it further if anyone has any ideas.

**Comment by timkelty** [comment]
Looking through the release notes, my only guess is something to do with the pagination changes, as that is exposed through these element connections. (‘pageInfo’)

Hopefully will have some time to look into it this week.

**Comment by timkelty** [comment]
@brandonkelly hmmm does appear pagination related…the actual thrown exception:

```
2019-03-20 07:32:28 [-][-][-][error][yii\base\UnknownPropertyException] yii\base\UnknownPropertyException: Setting unknown property: craft\web\twig\variables\Paginate::limit in /app/vendor/composer/yiisoft/yii2/base/BaseObject.php:163
```

**Comment by brandonkelly** [comment]
@timkelty What is the stack trace?

**Comment by timkelty** [comment]
nothing helpful - still digging: ```2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/exception.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/yiisoft/yii2/views/errorHandler/callStackItem.php
2019-03-20 07:32:28 [-][-][-][trace][yii\base\View::renderFile] Rendering view file: /app/vendor/composer/y```

**Comment by timkelty** [comment]
@brandonkelly Looking at the CrafQL code, I suspect its coming from ```\craft\helpers\Template::paginateCriteria($criteria)```

**Comment by brandonkelly** [comment]
Huh… `$pageInfo` is a `craft\web\twig\variables\Paginate` object there, and that has never had a `$limit` property.

Here’s all of the changes made to that class in 2.0.19:

https://github.com/craftcms/cms/compare/3.1.18...3.1.19#diff-d85eed56af7711fc3697dcae8e501448

So maybe this is just because we now have it extend `yii\base\BaseObject`, which is maybe more strict about setting unknown properties than vanilla classes are.

@markhuot Do you recall what the point of this line is?

https://github.com/markhuot/craftql/blob/e4b5a435a312539d455d72e2a0a2987f91ff660e/src/Types/Query.php#L120

**Comment by timkelty** [comment]
Yeah - seems like the intention is for the limit (default 100) to be on the `$criteria`, not `$pageInfo`.

**Comment by brandonkelly** [comment]
@markhuot Just submitted #249 to fix this, but I’m guessing all that is going to do is start limiting the results where previously they weren’t being limited at all, since `limit` wasn’t getting set on the element query like it would need to. So probably worth testing and making sure pagination works correctly.

**Comment by timkelty** [comment]
Thanks @brandonkelly!

**Comment by danireginatto** [comment]
+1 have a solution for this problem?

**Comment by swthate** [comment]
I'm interested in the progress of this as well. I'll have to hold some projects back to 3.1.18 until there is a fix as they rely on CraftQL 2 beta.

**Comment by markhuot** [comment]
Will be reviewing this, this week, to get a release cut.

**Comment by danireginatto** [comment]
Can we help in any way?


Daniel Reginatto
daniel@reginatto.com

**Comment by danireginatto** [comment]
> Will be reviewing this, this week, to get a release cut.

Mark, Can you tell us when we can test? My system is broke.

**Comment by brandonkelly** [comment]
In the meantime you should be able to get it working (and help test) by switching over to my PR branch.

To do that, add my repo to the `repositories` config in composer.json, and change your `markhuot/craftql` requirement to `dev-248-fix`:

```json
{
  "require": {
    "markhuot/craftql": "dev-248-fix",
    "...": "..."
  },
  "repositories": [
    {"type": "vcs", "url": "https://github.com/brandonkelly/craftql"}
  ],
  "...": "..."
}
```

Then run `composer update`.

**Comment by narration-sd** [comment]
That's a great contributive move,, @brandonkelly -- and a great method to keep in mind, seeing how it works....

**Comment by nc2-jordansmith** [comment]
Tested the `dev-248-fix` on Craft v3.1.19 and v3.1.20.1 and it works for me. I'll still need to wait for the release version for my team to use it, but thanks, @brandonkelly!

**Comment by dineufeld** [comment]
When to expect with an update? I have I an external API managed by CraftQL and I don't have an access to make downgrade of the craft version on the server.

**Comment by seamofreality** [comment]
Tested the dev-248-fix too (on Craft 3.1.20.1), got this error:

```
{
  "error": "Argument 1 passed to craft\\db\\Paginator::__construct() must implement interface yii\\db\\QueryInterface, array given, called in /usr/share/nginx/vendor/markhuot/craftql/src/Types/CategoryInterface.php on line 33"
}
```

Trying to query categories and their child entries:

```
{
  categories(group: teamCategory) {
    id
    title
    uri
    childrenConnection {
      edges {
        node {
          id
        }
      }
    }
  }
}
```

**Comment by seamofreality** [comment]
@brandonkelly Thanks a dozen for the quick fix, it seems to work now!
